account_invoice_report_grouped_by_picking/tests/test_account_invoice_group_picking.py

```diff
@@ -0,0 +1,294 @@
+                "default_code": "TESTPROD01",
+                "invoice_policy": "delivery",
+            }
+        )
+        cls.service = cls.env["product.product"].create(
+            {
+                "name": "Test service product",
+                "type": "service",
+                "invoice_policy": "order",
+            }
+        )
+        cls.partner = cls.env["res.partner"].create({"name": "Partner for test"})
+        cls.sale = cls.env["sale.order"].create(
+            {
+                "partner_id": cls.partner.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": cls.product.name,
+                            "product_id": cls.product.id,
+                            "product_uom_qty": 2,
+                            "product_uom": cls.product.uom_id.id,
+                            "price_unit": 100.0,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "name": cls.service.name,
+                            "product_id": cls.service.id,
+                            "product_uom_qty": 3,
+                            "product_uom": cls.service.uom_id.id,
+                            "price_unit": 50.0,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def get_return_picking_wizard(self, picking):
+        stock_return_picking_form = Form(
+            self.env["stock.return.picking"].with_context(
+                active_ids=picking.ids,
+                active_id=picking.ids[0],
+                active_model="stock.picking",
+            )
+        )
+        return stock_return_picking_form.save()
+
+    def test_account_invoice_group_picking(self):
+        # confirm quotation
+        self.sale.action_confirm()
+        # deliver lines2
+        self.sale.picking_ids[:1].action_confirm()
+        self.sale.picking_ids[:1].move_line_ids.write({"qty_done": 1})
+        self.sale.picking_ids[:1]._action_done()
+        # create another sale
+        self.sale2 = self.sale.copy()
+        self.sale2.order_line[:1].product_uom_qty = 4
+        self.sale2.order_line[:1].price_unit = 50.0
+        # confirm new quotation
+        self.sale2.action_confirm()
+        self.sale2.picking_ids[:1].action_confirm()
+        self.sale2.picking_ids[:1].move_line_ids.write({"qty_done": 1})
+        self.sale2.picking_ids[:1]._action_done()
+        sales = self.sale | self.sale2
+        # invoice sales
+        invoice = sales._create_invoices()
+        # Test directly grouping method
+        groups = invoice.lines_grouped_by_picking()
+        self.assertEqual(len(groups), 4)
+        self.assertEqual(groups[0]["picking"], groups[1]["picking"])
+        self.assertEqual(groups[2]["picking"], groups[3]["picking"])
```

Comment:
This is the test to generate the error above.

```suggestion
        self.assertEqual(groups[2]["picking"], groups[3]["picking"])
        # mix with invoice line that doesn't have neither move_line_ids
        # nor sale_line_ids
        vals = [
            {
                "name": "test invoice line",
                "move_id": invoice.id,
                "price_unit": 10.0,
                "account_id": invoice.invoice_line_ids[0].account_id.id,
            }
        ]
        lines = invoice.invoice_line_ids.create(vals)
```